src/test/java/it/gov/pagopa/reward/service/synchronous/RewardTrxSynchronousApiServiceImplTest.java

```diff
@@ -0,0 +1,175 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.util.Pair;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.Map;
+
+@ExtendWith(MockitoExtension.class)
+class RewardTrxSynchronousApiServiceImplTest {
+
+    @Mock
+    RewardContextHolderService rewardContextHolderServiceMock;
+    @Mock
+    OnboardedInitiativesService onboardedInitiativesServiceMock;
+    @Mock
+    InitiativesEvaluatorFacadeService initiativesEvaluatorFacadeServiceMock;
+    @Mock
+    UserInitiativeCountersRepository userInitiativeCountersRepositoryMock;
+    @Mock
+    SynchronousTransactionRequestDTOt2TrxDtoOrResponseMapper synchronousTransactionRequestDTOt2TrxDtoOrResponseMapperMock;
+    @Mock
+    RewardTransaction2SynchronousTransactionResponseDTOMapper rewardTransaction2SynchronousTransactionResponseDTOMapperMock;
+
+
+    @Test
+    void postTransactionPreviewNotOnboarded(){
+        // Given
+        SynchronousTransactionRequestDTO previewRequest = SynchronousTransactionRequestDTOFaker.mockInstance(1);
+        String initiativeId = "INITIATIVEID";
+
+        TransactionDTO transactionDTOMock = TransactionDTOFaker.mockInstance(1);
+        Mockito.when(synchronousTransactionRequestDTOt2TrxDtoOrResponseMapperMock.apply(Mockito.same(previewRequest))).thenReturn(transactionDTOMock);
+
+        InitiativeConfig initiativeConfig = InitiativeConfig.builder()
+                .initiativeId(initiativeId)
+                .build();
+        Mockito.when(rewardContextHolderServiceMock.getInitiativeConfig(initiativeId)).thenReturn(Mono.just(initiativeConfig));
+        Mockito.when(onboardedInitiativesServiceMock.isOnboarded(Mockito.same(transactionDTOMock.getHpan()), Mockito.same(transactionDTOMock.getTrxDate()), Mockito.same(initiativeId))).thenReturn(Mono.just(Boolean.FALSE));
+
+        SynchronousTransactionResponseDTO response = SynchronousTransactionResponseDTOFaker.mockInstanceBuilder(1)
+                        .initiativeId(initiativeId)
+                        .status(RewardConstants.REWARD_STATE_REJECTED)
+                        .rejectionReasons(List.of(RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE))
+                        .build();
+        Mockito.when(synchronousTransactionRequestDTOt2TrxDtoOrResponseMapperMock.apply(previewRequest, initiativeId, List.of(RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE))).thenReturn(response);
+        RewardTrxSynchronousApiServiceImpl rewardTrxSynchronousApiApiService = new RewardTrxSynchronousApiServiceImpl(rewardContextHolderServiceMock, onboardedInitiativesServiceMock,initiativesEvaluatorFacadeServiceMock, userInitiativeCountersRepositoryMock, synchronousTransactionRequestDTOt2TrxDtoOrResponseMapperMock, rewardTransaction2SynchronousTransactionResponseDTOMapperMock);
+
+        // When
+        try {
+            rewardTrxSynchronousApiApiService.previewTransaction(previewRequest, initiativeId).block();
+            Assertions.fail();
```

Comment:
let's provide a message to simplify the error identification
```suggestion
            Assertions.fail("Expected an Exception");
```